src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/RazorServiceBase.cs

```diff
@@ -1,48 +1,48 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+#nullable enable
+
 using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Razor.Language;
 using Microsoft.CodeAnalysis.Razor;
 using Microsoft.CodeAnalysis.Razor.ProjectSystem;
-using Newtonsoft.Json;
+using Microsoft.ServiceHub.Framework;
 
 namespace Microsoft.CodeAnalysis.Remote.Razor
 {
-    internal abstract class RazorServiceBase : ServiceHubServiceBase
+    internal abstract class RazorServiceBase : IDisposable
     {
-        public RazorServiceBase(Stream stream, IServiceProvider serviceProvider)
-            : base(serviceProvider, stream, GetRazorConverters())
+        protected readonly ServiceBrokerClient ServiceBrokerClient;
+
+        public RazorServiceBase(IServiceBroker serviceBroker)
         {
             RazorServices = new RazorServices();
 
-            // Due to this issue - https://github.com/dotnet/roslyn/issues/16900#issuecomment-277378950
-            // We need to manually start the RPC connection. Otherwise we'd be opting ourselves into 
-            // race condition prone call paths.
-            StartService();
+#pragma warning disable VSTHRD012 // Provide JoinableTaskFactory where allowed
+            ServiceBrokerClient = new ServiceBrokerClient(serviceBroker);
+#pragma warning restore
         }
 
         protected RazorServices RazorServices { get; }
 
+        public void Dispose()
+        {
+            ServiceBrokerClient.Dispose();
+        }
+
         protected virtual Task<ProjectSnapshot> GetProjectSnapshotAsync(ProjectSnapshotHandle projectHandle, CancellationToken cancellationToken)
         {
             if (projectHandle == null)
             {
                 throw new ArgumentNullException(nameof(projectHandle));
             }
 
-            return Task.FromResult<ProjectSnapshot>(new SerializedProjectSnapshot(projectHandle.FilePath, projectHandle.Configuration, projectHandle.RootNamespace));
-        }
-
-        private static IEnumerable<JsonConverter> GetRazorConverters()
-        {
-            var collection = new JsonConverterCollection();
-            collection.RegisterRazorConverters();
-            return collection;
+            var snapshot = new SerializedProjectSnapshot(projectHandle.FilePath, projectHandle.Configuration, projectHandle.RootNamespace);
+            return Task.FromResult<ProjectSnapshot>(snapshot);
```

Comment:
We don't inherit anything but `IDisposable` anymore, is there a reason we have have to return a task from this method? Is it just that we expect to use async API's inside it in the future and want to be ready?

Reply:
Ya potentially we may want to resolve the "actual" project. Leaving it async to future proof ourselves for that. At least for now